Mark the EVP_PKEY_METHOD arg as const on some EVP_PKEY_meth_get_*() funcs

mattcaswell opened the pull request:
Most EVP_PKEY_meth_get_*() functions mark the EVP_PKEY_METHOD argument as
const. But 3 did not. We fix those to be consistent.

This is pulled out of #16118 because it is an API change and requires a vote. This means #16118 will need to cast away a const for now. Once #16118 I will update this PR to remove those casts.

romen reviewed (approved):
The PR looks good to me.

I was favorable to have this even when it was part of the pkey method fix PR, I just noticed the change int he public API and raised the issue of potentially requiring a vote.

@mattcaswell do you plan to open the vote by email or to wait until the next OTC meeting?

mattcaswell commented:
> do you plan to open the vote by email or to wait until the next OTC meeting?

I was going to wait and I was hoping the vote could be held and closed in the meeting. If all approvals are done then I can just immediately push (assuming the vote passes).

t-j-h commented:
> > do you plan to open the vote by email or to wait until the next OTC meeting?
> > I was going to wait 

@mattcaswell I suggest you kick off the vote via email now as there is no need to wait on this or treat it as an exception to the vote process.

mattcaswell commented:
> @mattcaswell I suggest you kick off the vote via email now as there is no need to wait on this or treat it as an exception to the vote process.

Ok. I will hold off a few hours though until #16118 is merged. Once merged I will update this PR to remove the cast that it needed to add because of the const problem fixed here.

mattcaswell commented:
Rebased on latest master now that #16118 has been merged, and updated to clear away the now unnecessary cast that that PR introduced. Please reconfirm.

openssl-machine commented:
24 hours has passed since 'approval: done' was set, but as this PR has been updated in that time the label 'approval: ready to merge' is not being automatically set.  Please review the updates and set the label manually.

mattcaswell commented:
OTC: This is ok for merge to 3.0

mattcaswell commented:
Pushed. Thanks.
